great_expectations/data_context/store/store.py

```diff
@@ -102,42 +102,46 @@ def store_backend_id(self) -> str:
         return self._store_backend.store_backend_id
 
     @property
-    def key_class(self):
+    def key_class(self) -> Type[DataContextKey]:
         if self.ge_cloud_mode:
             return GeCloudIdentifier
         return self._key_class
 
     @property
-    def store_backend_id_warnings_suppressed(self):
+    def store_backend_id_warnings_suppressed(self) -> str:
         """
         Report the store_backend_id of the currently-configured StoreBackend, suppressing warnings for invalid configurations.
         Returns:
             store_backend_id which is a UUID(version=4)
         """
         return self._store_backend.store_backend_id_warnings_suppressed
 
+    @property
+    def config(self) -> dict:
```

Comment:
I'm honestly not sure haha - should this be changed?

Reply:
I don't know if I care too strongly but there is something to be said about being consistent in the code-base. Also, an ABC class will die at instantiation instead of when the unimplemented method is called so we know earlier if there is an error. I thought there was static analysis tools that could catch the instantiation of an abstract class but a quick google doesn't reveal this to me. If we could catch it via static analysis, I would have a stronger preference to make it an ABC.

Reply:
Pylint can catch this.